src/requirements.cpp

```diff
@@ -125,7 +125,13 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == std::numeric_limits<int>::max() ) {
```

Comment:
It's probably better to use `item::INFINITE_CHARGES` here for consistency with the rest of the crafting code.